packages/sirius-web/frontend/sirius-web-application/src/modals/update-library/UpdateLibraryTable.tsx

```diff
@@ -0,0 +1,132 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import Typography from '@mui/material/Typography';
+import Chip from '@mui/material/Chip';
+import {
+  MaterialReactTable,
+  MRT_ColumnDef,
+  MRT_PaginationState,
+  MRT_RowSelectionState,
+  useMaterialReactTable,
+} from 'material-react-table';
+import { useState, useEffect, useMemo } from 'react';
+import { GQLLibrary } from '../../views/library-browser/useLibraries.types';
+import { UpdateLibraryTableProps, UpdateLibraryTableState } from './UpdateLibraryTable.types';
+import { useLibraryVersions } from './useLibraryVersions';
+import { GQLLibraryVersion } from './useLibraryVersions.types';
+
+export const UpdateLibraryTable = ({ namespace, name, version, onSelectionChange }: UpdateLibraryTableProps) => {
+  const [state, setState] = useState<UpdateLibraryTableState>({
+    data: null,
+  });
+  const [pagination, setPagination] = useState<MRT_PaginationState>({
+    pageIndex: 0,
+    pageSize: 20,
+  });
+
+  const rows: GQLLibraryVersion[] = state.data?.viewer.library.versions.edges.map((edge) => edge.node) || [];
+  const count: number = state.data?.viewer.library.versions.pageInfo.count ?? 0;
+
+  const { data } = useLibraryVersions(namespace, name, version, pagination.pageIndex, pagination.pageSize);
+
+  useEffect(() => {
+    if (data) {
+      setState((prevState) => ({ ...prevState, data }));
+    }
+  }, [data]);
```

Comment:
Like I said in your previous PR, you do not need to store the result of a query/mutation in a state